to.etc.domui/src/main/java/to/etc/domui/util/exporters/ExcelWriterUtil.java

```diff
@@ -0,0 +1,223 @@
+package to.etc.domui.util.exporters;
+
+import org.apache.poi.ss.usermodel.BorderStyle;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.FillPatternType;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.usermodel.HorizontalAlignment;
+import org.apache.poi.ss.usermodel.IndexedColors;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFCellStyle;
+import org.apache.poi.xssf.usermodel.XSSFColor;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utils around cells styles and fonts, since these should be made cached and reusable inside workbook.
+ */
+@NonNullByDefault
+public class ExcelWriterUtil {
+
+	public enum FontStyle { BOLD("B"), ITALIC("I"), UNDERLINE("U"), Strikeout("S");
+
+		private String m_code;
+		FontStyle(String code) {
+			m_code = code;
+		}
+		public String getCode() {
+			return m_code;
+		}
+	}
+
+	private final ExcelFormat m_format;
+
+	private final Workbook m_workbook;
+
+	private final Map<String, CellStyle> m_styles = new HashMap<>();
+
+	private final Map<String, Font> m_fonts = new HashMap<>();
+
+	public ExcelWriterUtil(ExcelFormat format, Workbook workbook) {
+		m_format = format;
+		m_workbook = workbook;
+	}
+
+	private boolean m_autoSizeCols = true;
+
+	public CellStyle errorCs() {
+		String key = "error";
+		CellStyle cs = style(key);
+		if (null == cs) {
+			cs = m_workbook.createCellStyle();
+			cs.setAlignment(HorizontalAlignment.LEFT);
+			cs.setIndention((short) 1);
+			cs.setFillForegroundColor(IndexedColors.RED.getIndex());
+			cs.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+			cs.setWrapText(false);
+			Font font = cloneFromDefault();
+			font.setItalic(true);
+			font.setColor(IndexedColors.DARK_RED.getIndex());
+			cs.setFont(font);
+			m_styles.put(key, cs);
+		}
+		return cs;
+	}
+
+	/**
+	 * Default style with default font.
+	 * @return
+	 */
+	public CellStyle defaultCs() {
+		String key = "default";
+		CellStyle cs = m_styles.get(key);
+		if (null == cs) {
+			cs = createCellStyle();
+			cs.setFont(cloneFromDefault());
+			addStyle(key, cs);
+		}
+		return cs;
+	}
+
+	/**
+	 * Default style with bold font.
+	 * @return
+	 */
+	public CellStyle boldCs() {
+		String key = "bold";
+		CellStyle cs = m_styles.get(key);
+		if (null == cs) {
+			cs = createCellStyle();
+			cs.setFont(boldFont());
+			addStyle(key, cs);
+		}
+		return cs;
+	}
+
+	/**
+	 * Custom style, enables setting the rich color (awt) in case of xssf model in use, or fallback to predefined indexed color in case of less rich hssf model.
+	 * Also enables standard font decoration and setting the wrap text option.
+	 * @param xssfColor
```

Comment:
Do not add empty javadoc placeholders. They add nothing. You can switch off the automatic generation of these in the IDE.